pkg/logging/logging.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2021 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package fybriklogging
+
+import (
+	"encoding/json"
+
+	"github.com/go-logr/logr"
+)
+
+// LogType indicates the types of log entries supported.  Logging packages typically provide separate function
+// calls for the different log types.
+type LogType string
+
+const (
+	ERROR LogType = "error" // Errors encountered in the code
+	INFO  LogType = "info"  // Info logs are things you want to tell the user or administrator which are not errors.
+)
+
+// VerbosityLevel allows one to decide how "chatty" you want your logs to be.
+type VerbosityLevel int
+
+const (
+	MANDATORY VerbosityLevel = 0
+	DEBUG     VerbosityLevel = 1
+	TRACE     VerbosityLevel = 2
+)
+
+// LogEntry defines the information for a single entry stored in the system logs.
+// It is assumed that the timestamp is added automatically and thus not included in the log entry structure.
+// Verbosity and log type are added by the log library via which the entries are sent to stdout or stderr.
+type LogEntry struct {
+	// Caller indicates the cluster, component and function that generated the message in the format "cluster/component/function"
+	// Use the buildCallerPath function to construct this.
+	// +required
+	Caller string `json:"caller"`
+
+	// Message being logged.
+	// +required
+	Message string `json:"message"`
+
+	// Unique id of the kubernetes fybrikapplication (Needs to be unique over time in order to support history and not just current workloads)
+	// +required
+	FybrikAppGUID string `json:"fybrikAppGuid"`
+
+	// ForUser should be true if the message should be shared with the end user in the FybrikApplication status.
+	// Assumed false if not indicated.
+	// +optional
+	ForUser bool `json:"forUser,omitempty"`
+
+	// ForArchive should be true if the message should be archived long term.
+	// For example, contains full contents of FybrikApplication and its status and should be stored for auditing purposes.
+	// Assumed false if not indicated
+	// +optional
+	ForArchive bool `json:"forArchive,omitempty"`
+
+	// DataSetID is a unique identifier for the data set.
+	// It should include an indicator of the catalog from which it came, as well as the catalog id.
+	// +optional
+	DataSetID string `json:"datasetID,omitempty"`
+
+	// ResponseTime of the current operation in milliseconds.
+	// This is an optional parameter that can use in monitoring dashboards such as Kibana.
+	// +optional
+	ResponseTime int64 `json:"responseTime,omitempty"`
+
+	// Action is the current operation being called.
+	// For example, “create_plotter” or “update_blueprint”.
+	// +optional
+	Action string `json:"action,omitempty"`
```

Comment:
I think it is redundant and will be a part of the message. For example, "failed to update blueprint: <error>"

Reply:
@shlomitk1 By action I meant things like create, update, delete.  These are relevant especially from within reconcile.